sycl/source/detail/graph_impl.hpp

```diff
@@ -945,7 +1018,9 @@ class exec_graph_impl {
   exec_graph_impl(sycl::context Context,
                   const std::shared_ptr<graph_impl> &GraphImpl)
       : MSchedule(), MGraphImpl(GraphImpl), MPiSyncPoints(), MContext(Context),
-        MRequirements(), MExecutionEvents() {}
+        MRequirements(), MExecutionEvents() {
+    duplicateNodes();
```

Comment:
The doxy comment could maybe say that we're creating a copy of original nodes, and why

Reply:
Done